mixer/pkg/api/grpcServer_test.go

```diff
@@ -37,7 +37,7 @@ import (
 )
 
 type preprocCallback func(ctx context.Context, requestBag attribute.Bag, responseBag *attribute.MutableBag) error
-type checkCallback func(ctx context.Context, requestBag attribute.Bag) (adapter.CheckResult, error)
+type checkCallback func(ctx context.Context, requestBag attribute.Bag) (adapter.CheckResult, *mixerpb.RouteDirective, error)
```

Comment:
Why not simply extend CheckResult to include a RouteDirective? Then these signatures wouldn't change.